### PR TITLE
Docs: add instructions on how to watch changes in a package

### PR DESCRIPTION
**Pull request by kaelig**
This seems to work on my machine, other people's tests appreciated!

### Tophatting

1. Run `yarn build --root packages/storybook-a11y-test`
1. Run `yalc publish packages/storybook-a11y-test`
1. Inside `web`, run `yalc link --no-pure @shopify/storybook-a11y-test`
1. Run this newly recommended watcher:
```console
npx nodemon --watch packages/storybook-a11y-test \
  --exec 'yarn build --root packages/storybook-a11y-test \
  && npx yalc publish --push packages/storybook-a11y-test' \
  --ignore packages/storybook-a11y-test/build \
  --ext ts --ignore '*.d.ts'
```
5. Make changes to the `storybook-a11y-test` package, and verify that they're applied in `web`.

Ideally, try this workflow with a couple of other packages, see if it works elsewhere!

### CODE REVIEW

**Comment by kaelig** [comment]
I agree that `yarn tophat storybook-a11y-test <other-packages...>` would be great!

File: package.json [hunk 1]
```diff
@@ -18,7 +18,9 @@
     "clean": "rimraf ./packages/*/build ./packages/*/*.{d.ts,js,esnext,mjs} .sewing-kit",
     "generate": "plop",
     "generate:package": "plop package && yarn plop docs",
-    "yalc": "yalc"
+    "yalc": "yalc",
+    "publish:local": "yarn build --root packages/$PKG_NAME && npx yalc publish --push packages/$PKG_NAME",
```
Comment (kaelig) [inline package.json:22]: Turns out `yalc` is listed in the `devDependencies`, so no need to use `npx` to run it – my bad!

```suggestion
    "publish:local": "yarn build --root packages/$PKG_NAME && yalc publish --push packages/$PKG_NAME",
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -40,26 +40,24 @@ The [documentation](../documentation) directory in this repo covers the more gra
 - [Resources](../documentation/resources.md): good resources for understanding this project’s tech stack.
 - [Getting started](../documentation/getting-started.md): some tools we recommend for getting the most out of this project.
 
-## Testing your changes in a local project
+## 🎩 in a local project
 
-You can use [`yalc`](https://github.com/wclr/yalc) to test local changes to quilt packages inside consuming repos without publishing new versions.
+You can use [`yalc`](https://github.com/wclr/yalc) to publish the pacakge and [`nodemon`](https://nodemon.io/) to watch the file changes.
```
Comment (kaelig) [inline .github/CONTRIBUTING.md:45]:
```suggestion
You can use [`yalc`](https://github.com/wclr/yalc) to publish the package and [`nodemon`](https://nodemon.io/) to watch the file changes.
```

File: package.json [hunk 3]
```diff
@@ -18,7 +18,9 @@
...
+    "tophat": "npx nodemon --watch packages/$PKG_NAME --exec 'yarn publish:local' --ignore packages/$PKG_NAME/build --ext ts --ignore '*.d.ts' --ignore '*.test.*'"
```
[the remark this answers is not in the document]
Comment (michenly) [reply package.json:23]: that structure would be ideal, yes. 
However, I do think what we have for now is better than nothing. While multiple packages support will be great as a follow up imprvement.

**Pull request merged**